### PR TITLE
CB-13601 Using CM hosts api instead of cluster api to get host list for decommission

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java [hunk 1]
```diff
@@ -196,9 +196,9 @@ private void addInstancesNotKnownByCMWithLimit(Integer limit, Set<InstanceMetaDa
         if (hostsToRemove.size() != hostNames.size()) {
             LOGGER.debug("Not all hosts found in the given host group. [{}, {}]", hostGroup.getName(), hostNames);
         }
-        ClustersResourceApi clustersResourceApi = clouderaManagerApiFactory.getClustersResourceApi(client);
+        HostsResourceApi hostsResourceApi = clouderaManagerApiFactory.getHostsResourceApi(client);
         try {
-            ApiHostList hostRefList = clustersResourceApi.listHosts(stack.getName(), null, null, null);
+            ApiHostList hostRefList = hostsResourceApi.readHosts(null, null, null);
```
Comment (horadla23) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java:201]: what happens if third param (view) is null? Will it return with full view? Or with simplest view?

Comment (biharitomi) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java:201]: What happens if CB sends a node to be decommissioned when it is not part of CM cluster?

Comment (horadla23) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java:201]: it seems it is handled here in the filter:
```
Sets.newHashSet(hostsToRemove.keySet()).stream()
                    .filter(hostName -> !runningHosts.contains(hostName))
                    .forEach(hostsToRemove::remove);
```

Comment (daszabo) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java]: I set it to SUMMARY just to be sure.

Comment (daszabo) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDecomissioner.java]: Decommission does not seem to be cluster based. It was able to decommission it with no problem.

**Review by horadla23 — APPROVED** [review]
LGTM

**Comment by daszabo** [comment]
Do not merge yet please. There will be another change to actually delete the hosts. I just noticed that decommission really just decommission the hosts, but keeps them is CM.

**Comment by daszabo** [comment]
> Do not merge yet please. There will be another change to actually delete the hosts. I just noticed that decommission really just decommission the hosts, but keeps them is CM.

DONE

**Review by biharitomi — APPROVED** [review]
If it is not a problem on CM side that a node is not part of the CM's cluster but decommission is triggered on it, then LGTM

**Pull request merged**